sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -314,7 +317,13 @@ class __SYCL_EXPORT modifiable_command_graph {
   /// @param path The path to write the DOT file to.
   /// @param verbose If true, print additional information about the nodes such
   /// as kernel args or memory access where applicable.
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  void print_graph(const std::string path, bool verbose = false) const {
```

Comment:
Even with the `__INTEL_PREVIEW_BREAKING_CHANGES` guard the public API for this method is still `std::string`, why does using `sycl::detail::string_view` in the implementation change the public ABI? Is it due to a C++ optimization?

Reply:
Let me clarify the purpose of this "ABI-neutrality"
This is needed for our big customers who need to use dual ABI.
Refer to https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html
In summary, when the customer sets _GLIBCXX_USE_CXX11_ABI=0, ABI incompatibility prevents them from using SYCL.
So, we have made lots of effort to make our SYCL ABI-neutral.
The public API remains the same as before since the header will be included in the user code.
But, ABI between the header and our library needs to be ABI neutral.
That's why we need this changeset.

Reply:
ah that's interesting, thanks for the details